source/common/buffer/zero_copy_input_stream_impl.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "common/buffer/buffer_impl.h"
+
+#include "google/protobuf/io/zero_copy_stream.h"
+
+namespace Envoy {
+
+namespace Buffer {
+
+class ZeroCopyInputStreamImpl : public virtual google::protobuf::io::ZeroCopyInputStream {
+public:
+  // Create input stream with one buffer, and finish immediately
```

Comment:
Seems all usage are using InstancePtr, so just changed the signature with InstancePtr in this class.